lighthouse-viewer/app/src/drag-and-drop.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * @license Copyright 2016 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * Manages drag and drop file input for the page.
+ */
+class DragAndDrop {
```

Comment:
slightly simpler than the old FileUploader. Handles only drag and dropping of files (the hidden input is now just in the page and its handler and `readFile` are now in the viewer main class)